ts/context.ts

```diff
@@ -412,6 +412,7 @@ export class QuickJSContext implements LowLevelJavascriptVm<QuickJSHandle>, Disp
    */
   newFunction(name: string, fn: VmFunctionImplementation<QuickJSHandle>): QuickJSHandle {
     const fnId = ++this.fnNextId
+    // If fnID > 16777218 (24-bit max int) the map will error with too many keys
```

Comment:
If you want this to scale, you can use an array of maps or an array-of-arrays; use the high byte to pick the index of the array to store the map/sub-array, and the rest of the bytes as the key/index in the map/sub-array.

Reply:
That's a good idea. It still has an issue once the ids > Max 32-bit int, but that is quite a bit more space

Reply:
I've implemented this as a map of maps with some protected helper functions for maintenance. Seems to work for the existing cases